src/vi/comanhua/src/eu/kanade/tachiyomi/extension/vi/comanhua/CoManhua.kt

```diff
@@ -0,0 +1,164 @@
+package eu.kanade.tachiyomi.extension.vi.comanhua
+
+import eu.kanade.tachiyomi.multisrc.wpcomics.WPComics
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class CoManhua : WPComics(
+    "CoManhua",
+    "https://comanhuaz.com",
+    "vi",
+    gmtOffset = null,
+) {
+
+    private val coManhuaDateFormat = SimpleDateFormat("dd/MM/yyyy HH:mm", Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("Asia/Ho_Chi_Minh")
+    }
+
+    private fun parseChapterDate(dateString: String): Long {
+        return try {
+            coManhuaDateFormat.parse(dateString)?.time ?: 0L
+        } catch (e: ParseException) {
+            0L
+        }
+    }
+
+    override val client: OkHttpClient = super.client.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override val popularPath = "truyen-de-cu"
+
+    override fun popularMangaRequest(page: Int): Request {
+        val offset = (page - 1) * 40
+        val url = "$baseUrl/$popularPath?offset=$offset"
+        return GET(url, headers)
+    }
+
+    override fun popularMangaSelector() = "div.pda.manga-list div.manga-item"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.selectFirst("div.manga-title a")?.let {
+            title = it.text()
+            setUrlWithoutDomain(it.attr("abs:href"))
+        }
+        element.selectFirst("div.manga-image img")?.let { imageOrNull(it)?.let { url -> thumbnail_url = url } }
+    }
+
+    override fun popularMangaNextPageSelector() = "div.list-pagination a:last-child:not(.active)"
+
+    override val queryParam = "name"
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/tim-truyen".toHttpUrl().newBuilder()
```

Comment:
Override `searchPath` with `tim-truyen` (and use that here). This will also fix the genre fetcher.